rich/traceback.py

```diff
@@ -362,27 +395,35 @@ def safe_str(_object: Any) -> str:
             stacks.append(stack)
             append = stack.frames.append
 
+            exclude_patterns = exclude_locals if exclude_locals else ()
+
             for frame_summary, line_no in walk_tb(traceback):
                 filename = frame_summary.f_code.co_filename
                 if filename and not filename.startswith("<"):
                     if not os.path.isabs(filename):
                         filename = os.path.join(_IMPORT_CWD, filename)
                 if frame_summary.f_locals.get("_rich_traceback_omit", False):
                     continue
+
+                if show_locals:
+                    local_vars = {
```

Comment:
AFAICT This will entirely omit matching regexes, which might leave the reader wondering why some values are missing. It would be nicer if the value is replaced by something to indicate it was removed. So the user would see something along these lines:

```
     foo = "bar"
password = <redacted>
```

The best place to do this is probably in the `Pretty` class. This would also allow you to filter recursively so that a "password" key in a dict would also be redacted.

P.S. We have a policy of no abbreviations, in particular single letters.

Reply:
Gotcha! I'll have another go!

Reply:
So I didn't do the recursive implementation, I've taken quite a conservative approach. The output you get looks something like this:

```
    ╭────────────────────── locals ──────────────────────╮
    │         password = <redacted variable name>        │
    │          my_list = [1, 2, 3]                       │
    │ credentials_dict = <redacted variable content>     │
    ╰────────────────────────────────────────────────────╯
```

What are your thoughts?